sbnana/CAFAna/Core/SpectrumLoader.cxx

```diff
@@ -145,6 +193,41 @@ namespace ana
     } // end for n
   }
 
+  //----------------------------------------------------------------------
+  void SpectrumLoader::HandleFile(TFile* f, Progress* prog, unsigned int N)
+  {
+    assert(!f->IsZombie());
+
+    TTree* tr = (TTree*)f->Get("recTree");
+    assert(tr);
+
+    // We try to access this field for every record. It was only added to the
+    // files in late 2021, and we don't want to render all earlier files
+    // unusable at a stroke. This logic can safely be removed once all extant
+    // files have such a field (estimate mid-2022?)
+    const bool has_husk = tr->GetLeaf("rec.hdr.husk");
+
+    caf::SRSpillProxy sr(tr, "rec");
+
+    //    FloatingExceptionOnNaN fpnan;
+
+    tr->LoadTree(N); // Only want a specific event from the TTree
+
+    // If there is no husk field there is no concept of husk events
+    if(!has_husk) sr.hdr.husk = false;
+
+    HandleRecord(&sr);
```

Comment:
How does this hook into the event display use-case? Right now you are just filling exactly one event into the existing spectra etc. Do you subclass SpectrumLoader to override what HandleRecord does?